modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -4028,21 +4028,12 @@ def sort_columns_by_row_values(self, rows, ascending=True, **kwargs):
 
     # Cat operations
     def cat_codes(self):
-        def func(df) -> np.ndarray:
-            # `df` is supposed to be consisted of multiple partitions,
-            # which should be concatenated before applying a function.
-            # `pd.concat` doesn't preserve categorical dtype
-            # if the dfs have categorical columns
-            # so we intentionaly restore the right dtype.
-            # TODO: revert the change when https://github.com/pandas-dev/pandas/issues/51362 is fixed.
+        def func(df: pandas.DataFrame) -> pandas.DataFrame:
             ser = df.iloc[:, 0]
-            if ser.dtype != "category":
-                ser = ser.astype("category", copy=False)
```

Comment:
since now, we shouldn't do this cast as disappearance of the categorical dtype will mean that we lost valid codes and should report an error